gh-127691: add type checks when using `PyUnicodeError` objects (#127694)
python/cpython: Misc/NEWS.d/next/C_API/2024-12-06-16-53-34.gh-issue-127691.k_Jitp.rst

The :ref:`Unicode Exception Objects <unicodeexceptions>` C API

> now raises :exc:`TypeError` if its exception argument is not an

Author: ```suggestion
now raises a :exc:`TypeError` if its exception argument is not a
```

> The :ref:`Unicode Exception Objects <unicodeexceptions>` C API
> now raises a :exc:`TypeError` if its exception argument is not a
> :exc:`UnicodeError`. Patch by Bénédikt Tran.

Author: ```suggestion
The :ref:`Unicode Exception Objects <unicodeexceptions>` C API
now raises a :exc:`TypeError` if its exception argument is not
a :exc:`UnicodeError` object. Patch by Bénédikt Tran.
```

> The :ref:`Unicode Exception Objects <unicodeexceptions>` C API now
> type-checks its exception argument, raising a :exc:`TypeError` if
> the latter is not an instance of :exc:`UnicodeError`.
> Patch by Bénédikt Tran.

Author: ```suggestion
The :ref:`Unicode Exception Objects <unicodeexceptions>` C API
now raises :exc:`TypeError` if its exception argument is not an
:exc:`UnicodeError`. Patch by Bénédikt Tran.
```
